pkg/cloudprovider/aws/block_store.go

```diff
@@ -218,6 +219,18 @@ func (b *blockStore) DeleteSnapshot(snapshotID string) error {
 	}
 
 	_, err := b.ec2.DeleteSnapshot(req)
+	if err == nil {
+		return nil
+	}
+
+	if awsErr, ok := err.(awserr.Error); ok {
+		// if it's a NotFound error, we don't need to return an error
+		// since the snapshot is not there.
+		// see https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html
+		if awsErr.Code() == "InvalidSnapshot.NotFound" {
+			return nil
+		}
+	}
```

Comment:
A bit of a nit comment incoming :)

I would nest the code in line 226 inside a `if err != nil`  and after that, still inside that if, return `errors.WithStack(err)`, so that the very last line of this func would be `return nil`.  This mainly because it causes least surprise but the reasoning I think is that it's more important to read the error conditions so if all of that come first/at the top it is easier to read/find.